plugins/module_utils/identity/bitwarden.py

```diff
@@ -0,0 +1,321 @@
+
+        # Filter by folder and attribute since `bw` ORs filters together.
+        matches = [Item(dict_=d) for d in Client.run_json(query)
+                   if self.folder.matches(d) and d[attribute] == value]
+
+        if not matches:
+            raise KeyError('Item not found')
+        if len(matches) > 1:
+            raise AssertionError('Item not unique')
+
+        return matches[0]
+
+    def _dict_resolve_callback(self, value):
+        self._organization = Organization(id_=value['organizationId'])
+        self._folder = Folder(id_=value['folderId'])
+
+
+class Folder(_Element):
+    '''A folder in the password database.'''
+
+    organization = property(lambda self: self._organization)
+
+    def __init__(self, name=None, id_=None, organization=None, dict_=None):
+        if [name is None, id_ is None, dict_ is None].count(False) > 1:
+            raise AssertionError('Cannot specify folder in multiple ways')
+
+        self._name = name
+        self._id = id_
+        self._organization = organization
+
+        if dict_:
+            self.dict = dict_
+
+    def _by_attribute(self, attribute, value):
+        '''Return folder dict where the specified attribute
+        has the specified value.
+        '''
+        matches = [folder for folder in self.organization.folders if folder.dict[attribute] == value]
+
+        if not matches:
+            raise KeyError("Folder not found")
+        if len(matches) > 1:
+            raise AssertionError("Folder not unique")
+
+        return matches[0]
+
+    @property
+    def items(self):
+        query = ['list', 'items']
+
+        # Only restrict by folder if specified.
+        if self.name is not None:
+            # bw treats 'null' as items not belonging to any folder.
+            query.extend(['--folderid', self.id or 'null'])
+
+        # Filter by organization since `bw` ORs filters together.
+        return [Item(dict_=d) for d in Client.run_json(query) if self.organization.matches(d)]
+
+    def get_item(self, name=None, id_=None):
+        return Item(name, id_, self, self.organization)
+
+    def matches(self, folder):
+        '''Return whether folder matches a pattern indicated by self.
+
+        If self.name is None, any folder matches.
+        If self.name is '' or 'No Folder', only the null folder is matched.
+        All other folders are matched by id.
+        '''
+
+        if self.name is None:
+            return True
+
+        # No Folder is a special folder name in bw.
+        if self.name in ('', 'No Folder'):
+            return folder['folderId'] is None
+
+        return self.id == folder['folderId']
+
+
+class Organization(_Element):
+    '''An organization in the password database.'''
+
+    def __init__(self, name=None, id_=None, dict_=None):
+        if [name is None, id_ is None, dict_ is None].count(False) > 1:
+            raise AssertionError('Cannot specify organization in multiple ways')
```

Comment:
🤔 I think I had this before (since I would normally write an `assert` myself, not `raise AssertionError`), but IIRC, this was explicitly flagged by one of the sanity checkers for this project and indicted `raise AssertionError` as the proposed alternative.

Reply:
I didi not know the sanity checker would pick on that, but it kinda makes sense. Again, I think you should be using a different exception, but I will not press the point.